### PR TITLE
Introduce Turbolinks to improve navigation behavior

### LINKED ISSUE #665: Improve Auth UI navigation handling
We may want to use other battle-tested libraries to handle browser history & navigation in Auth UI.

ref: https://github.com/turbolinks/turbolinks

### LINKED ISSUE #668: Press back button on Connected Identities page cannot back to main setting page
Step to reproduce:
1. Signup an account with the application have SSO configured
2. Go to Connected Identities
3. Try connecting to a SSO that will result login failed
4. Press back button on the Connected Identities page will back to the SSO page instead of back to the previous Authgear setting page

### PR DESCRIPTION
**Pull request by kiootic**
fix #665 
fix #668

### CODE REVIEW

File: authui/src/authgear.ts [hunk 1]
```diff
@@ -8,6 +8,15 @@ window.api.onLoad(() => {
   document.body.classList.add("js");
 });
 
+// Handle history tracking.
+
+let inHistorySettings = false;
+window.api.onLoad(() => {
+  if (window.location.pathname === "/settings") {
```
Comment (louischan-oursky) [inline authui/src/authgear.ts:15]: I just realized `/change_password` is not under `/settings`.

**Review by louischan-oursky — COMMENTED** [review]
Now we have to build the static asset to make the server functional. Do you have any suggested workflow? Can we document it in README or even include it in Makefile?

File: authui/src/authgear.ts (hunk 1 above)
Comment (kiootic) [reply authui/src/authgear.ts:15]: I suppose we can move it under settings, so it'll be `/settings/change_password` and `/settings/change_secondary_password`.

I just remembered `/enter_login_id`, its name looks like part of interaction flow, yet it's part of settings page. Should we change its name too?

**Comment by kiootic** [comment]
> Now we have to build the static asset to make the server functional. Do you have any suggested workflow? Can we document it in README or even include it in Makefile?

I suppose we can invoke `$(MAKE) static` in `make vendor` script to make setup more convenient.

File: authui/src/authgear.ts (hunk 1 above)
Comment (louischan-oursky) [reply authui/src/authgear.ts:15]: > I suppose we can move it under settings, so it'll be /settings/change_password and /settings/change_secondary_password.

Agree.

> I just remembered /enter_login_id, its name looks like part of interaction flow, yet it's part of settings page. Should we change its name too?

Agree. Let's change from `/enter_login_id` to `/settings/enter_login_id`?

**Pull request merged**